chore: readme based on new template

alvarosabu opened the pull request:
This PR adds a new README.md based on the company OSS Readme template

alvarosabu committed:
docs: update README and add Type Generation Command documentation

- Enhanced the README with a new section on features, improving clarity on the capabilities of the Storyblok CLI.
- Added a comprehensive documentation file for the Type Generation Command, detailing prerequisites, usage examples, architecture, and testing strategies to ensure type safety in Storyblok components.
- Improved visual elements in the README, including badge colors and layout adjustments for better readability.

alvarosabu committed:
docs: clean up README formatting and improve readability

- Removed unnecessary blank lines and adjusted formatting in the main README and the Type Generation Command documentation to enhance clarity and consistency.
- Ensured proper alignment of bullet points for better visual presentation.

edodusi commented on src/commands/types/generate/README.md:
```diff
@@ -0,0 +1,320 @@
+  });
+
+// 2. Operations Layer (actions.ts)
+-generateTypes()
```
```suggestion
- generateTypes()
```

edodusi commented on src/commands/types/generate/README.md:
```diff
@@ -0,0 +1,320 @@
+
+Before generating types, ensure you have:
+
+- Access to the target Storyblok space
```
```suggestion
- A Storyblok account and an active login on the CLI
```

edodusi commented on src/commands/types/generate/README.md:
```diff
@@ -0,0 +1,320 @@
+Before generating types, ensure you have:
+
+- Access to the target Storyblok space
+- A Storyblok account with appropriate permissions
```
```suggestion
- Access to the target Storyblok space and proper permissions
```

edodusi commented on src/commands/types/generate/README.md:
```diff
@@ -0,0 +1,320 @@
+- Access to the target Storyblok space
+- A Storyblok account with appropriate permissions
+- TypeScript installed in your project (if you plan to use the generated types)
+- Pull components from the target space using the `storyblok components pull` command
```
```suggestion
- Components from the target space pulled using the `storyblok components pull` command
```

alvarosabu replied:
Hi @edodusi ignore the internal Readme, is gonna be removed anyways, the changes in this PR where barely linter based, the important one is the README.md in the core

alvarosabu committed:
docs: Update src/commands/types/generate/README.md

Co-authored-by: Edoardo Dusi <edo@edoardodusi.com>

alvarosabu committed:
docs: Update src/commands/types/generate/README.md

Co-authored-by: Edoardo Dusi <edo@edoardodusi.com>

alvarosabu committed:
docs: fix formatting in Type Generation Command section of README.md

- Removed unnecessary hyphen from the `generateTypes()` function reference to ensure consistency in formatting.
- This change improves the clarity of the documentation for users referencing the Type Generation Command.
